### PR TITLE
Estimators memory leak testing

### PR DESCRIPTION
**Pull request by Alexsandruss**
# Description
Changes proposed in this pull request:
- Move memory leak tests from unittest to pytest
- Enable memory leak testing for all patched estimators except TSNE

### CODE REVIEW

File: tests/test_memory_usage.py [hunk 1]
```diff
@@ -1,78 +1,134 @@
...
+    mem_before, _ = tracemalloc.get_traced_memory()
+    for train_index, test_index in kf.split(x):
+        if isinstance(x, np.ndarray):
+            x_train, x_test = x[train_index], x[test_index]
+            y_train, y_test = y[train_index], y[test_index]
+        elif isinstance(x, pd.core.frame.DataFrame):
+            x_train, x_test = x.iloc[train_index], x.iloc[test_index]
+            y_train, y_test = y.iloc[train_index], y.iloc[test_index]
+        alg = estimator()
+        alg.fit(x_train, y_train)
```
Comment (SmirnovEgorRu) [inline tests/test_memory_usage.py:114]: What about predict/transform?

File: daal4py/sklearn/_utils.py [hunk 2]
```diff
@@ -117,4 +117,6 @@ def get_dtype(X):
 
 def get_number_of_types(dataframe):
```
Comment (PetrovKP) [inline daal4py/sklearn/_utils.py:118]: This refers to the number of unique types in the pandas. You get size of all type.

Why did it become necessary to change this function?

Comment (Alexsandruss) [reply daal4py/sklearn/_utils.py:118]: Tests uncovered that when pd.Series is passed, error raises due to getting set from non-iterable np.dtype: `set(dtypes)`

Comment (PetrovKP) [reply daal4py/sklearn/_utils.py:118]: Ok, thanks :)

**Comment by PetrovKP** [comment]
Move pls this test here: https://github.com/IntelPython/daal4py/tree/master/daal4py/sklearn/test

**Comment by PetrovKP** [comment]
Failed in CI:
```
Estimator = <class 'daal4py.sklearn.linear_model._logistic_path_0_22.LogisticRegression'>
ata_transform_function = <function ndarray_c at 0x0000015D5FA73820>

   def _kfold_function_template(estimator, data_transform_function):
       tracemalloc.start()
   
       x, y, data_memory_size = gen_clsf_data()
       kf = KFold(n_splits=10)
       x, y = data_transform_function(x, y)
   
       mem_before, _ = tracemalloc.get_traced_memory()
       for train_index, test_index in kf.split(x):
           if isinstance(x, np.ndarray):
               x_train, x_test = x[train_index], x[test_index]
               y_train, y_test = y[train_index], y[test_index]
           elif isinstance(x, pd.core.frame.DataFrame):
               x_train, x_test = x.iloc[train_index], x.iloc[test_index]
               y_train, y_test = y.iloc[train_index], y.iloc[test_index]
           alg = estimator()
           alg.fit(x_train, y_train)
           if hasattr(alg, 'predict'):
               alg.predict(x_test)
           elif hasattr(alg, 'transform'):
               alg.transform(x_test)
           elif hasattr(alg, 'kneighbors'):
               alg.kneighbors(x_test)
       del alg, x_train, x_test, y_train, y_test
       mem_after, _ = tracemalloc.get_traced_memory()
       tracemalloc.stop()
       mem_diff = mem_after - mem_before
   
       assert mem_diff < 0.25 * data_memory_size, \
           'Size of extra allocated memory is greater than 25% of input data:' \
           f'\n\tInput data size: {data_memory_size} bytes' \
           f'\n\tExtra allocated memory size: {mem_diff} bytes' \
           f' / {round((mem_diff) / data_memory_size * 100, 2)} %'
       AssertionError: Size of extra allocated memory is greater than 25% of input data:
             Input data size: 808000 bytes
             Extra allocated memory size: 3067886 bytes / 379.69 %
       assert 3067886 < (0.25 * 808000)
```

**Comment by PetrovKP** [comment]
Do we have problems with logreg for memory tests?

**Comment by Alexsandruss** [comment]
930% memory usage overhead on all platforms and layouts (except C-ordered ndarray on Linux) probably shows that train data is copied each time.
Since all other estimators work normally, it's memory leak in LogisticRegression. Does it stops this PR from merging?

**Comment by PetrovKP** [comment]
> Does it stops this PR from merging?

No, but you need to disable the logreg test for now.

**Review by PetrovKP — APPROVED** [review]
Great work! Thanks you

**Pull request merged**